packages/react-components/src/components/Accordion/hooks.ts

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+
+interface UseAccordionProps {
+  isControlled: boolean;
+  isExpanded: boolean;
+  setShouldBeVisible: React.Dispatch<React.SetStateAction<boolean>>;
+  onOpen?: () => void;
+  onClose?: () => void;
+}
+
+interface IUseAccordion {
+  handleExpandChange: (isExpanded: boolean) => void;
+  handleKeyDown: (event: React.KeyboardEvent<HTMLDivElement>) => void;
+}
+
+export const useAccordion = ({
+  isControlled,
+  isExpanded,
+  setShouldBeVisible,
+  onOpen,
+  onClose,
+}: UseAccordionProps): IUseAccordion => {
+  const handleExpandChange = (isExpanded: boolean) => {
+    if (isExpanded) {
+      onClose?.();
+      !isControlled && setShouldBeVisible(false);
+    } else {
+      onOpen?.();
+      !isControlled && setShouldBeVisible(true);
```

Comment:
You do not need to duplicate the setter, use isExpanded value as an argument for this function